packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -103,30 +97,25 @@ export function CrossmintWalletProvider({
 
         try {
             setWalletState({ status: "in-progress" });
-            const signer = config?.signer ?? (await getPasskeySigner());
-            const wallet = await smartWalletSDK.getOrCreateWallet(
-                { jwt: crossmint.jwt as string },
-                defaultChain,
-                config ?? {
-                    signer,
-                },
-                {
-                    onWalletCreationFailed: createPasskeyPrompt("create-wallet-error"),
-                    onTransactionSigningStarted: () => {
-                        if (walletCache.isWalletInitialized) {
-                            return Promise.resolve();
-                        }
-                        return createPasskeyPrompt("transaction")();
-                    },
-                    onTransactionSigningFailed: createPasskeyPrompt("transaction-error"),
-                    onTransactionCompleted: () => {
-                        walletCache.setWalletInitialized(true);
-                        return Promise.resolve();
-                    },
+
+            if (props.type === "evm-smart-wallet") {
+                const evmArgs = props.args as WalletTypeToArgs["evm-smart-wallet"];
+                let wallet: EVMSmartWallet | null = null;
+                const walletArgs: WalletTypeToArgs["evm-smart-wallet"] = {
+                    chain: evmArgs.chain,
+                    adminSigner: evmArgs.adminSigner,
+                    linkedUser: evmArgs.linkedUser,
+                };
+                wallet = await smartWalletSDK.getOrCreateWallet("evm-smart-wallet", walletArgs);
```

Comment:
cant we always pass the adminSigner though and it just fetches the wallet if it exists?